scripts/jwst/build_parent_sample.py

```diff
@@ -0,0 +1,485 @@
+            #dec = row['dec']  # Assuming there is a 'dec' column
+
+            key = str(id)
+
+            
+
+            # Get the smallest shape among all images
+            stamp = JWST_multilambda[f][c]
+            s_x=stamp.shape[0]
+            s_y=stamp.shape[1]
+            #s_x = min([stamp.shape[0] for f in filter_list])
+            #s_y = min([stamp.shape[1] for f in filter_list])
+
+                # Raise a warning if one of the images has a different shape than 'smallest_shape'
+            #for f in filter_list:
+            #    if stamp.shape != (s_x, s_y):
+            #        #print(JWST_multilambda[f].shape)
+            #        print(f"The image for object {key} has a different shape depending on the band. It's the {n_problems+1}th time this happens.")
+            #        n_problems += 1
+            #        break
+
+                # Crop the images to the smallest shape
+            image = np.stack([
+                    JWST_multilambda[f][c][:s_x, :s_y].astype(np.float32) for f in filter_list
+                ], axis=0).astype(np.float32)
+                
+                # Cutout the center of the image to desired size
+            s = image.shape
+            #print(s)
+            center_x = s[1] // 2
+            start_x = center_x - _image_size // 2
+            center_y = s[2] // 2
+            start_y = center_y - _image_size // 2
+            image = image[:,
+                            start_x:start_x+_image_size, 
+                            start_y:start_y+_image_size]
+            assert image.shape == (len(filter_list),_image_size, _image_size), ("There was an error in reshaping the image to desired size. Probably a fiter is missing? Check the available list of filters for the survey", image.shape, s )
+
+                # Compute the PSF FWHM in arcsec
+                #psf_fwhm = []
+                #for f in _filters:
+                #    b = f.lower().split('-')[-1]
+                #    psf_mxx = row[f'{b}_sdssshape_psf_shape11']
+                #    psf_myy = row[f'{b}_sdssshape_psf_shape22']
+                #    psf_mxy = row[f'{b}_sdssshape_psf_shape12']
+                #    psf_fwhm.append(2.355 * (psf_mxx * psf_myy - psf_mxy**2)**(0.25)) # in arcsec
+                #psf_fwhm = np.array(psf_fwhm).astype(np.float32)
+
+                # Initialize the example with image data
+            #print(np.zeros((image.shape[1],image.shape[2])).shape)
+             # Automatically create _filters by formatting each entry in _filter_list (astropile nomenclature)
+            filters = [f'jwst_nircam_{filter_name}' for filter_name in filter_list]
+            images.append({
+                        'object_id': id,
+                        'image_band': np.array([f.lower().encode("utf-8") for f in filters], dtype=_utf8_filter_type),
+                        'image_array': image,
+                        'image_psf_fwhm': image,
```

Comment:
Can you compute the FWHM somehow? For space based it's fine to have a single number

Reply:
And you can add another entry for 'image_psf_array'